src/store/wizardSlice.ts

```diff
@@ -719,6 +778,53 @@ export const wizardSlice = createSlice({
     setFirstBootScript: (state, action: PayloadAction<string>) => {
       state.firstBoot.script = action.payload;
     },
+
+    addUser: (state) => {
+      const newUser = {
+        name: '',
+        password: '',
+        confirmPassword: '',
+        sshKey: '',
+        groups: [],
+        isAdministrator: false,
+      };
+
+      state.users.push(newUser);
+    },
+    setUserNameByIndex: (state, action: PayloadAction<UserPayload>) => {
+      state.users[action.payload.index].name = action.payload.name;
+    },
+    setUserPasswordByIndex: (
+      state,
+      action: PayloadAction<UserPasswordPayload>
+    ) => {
+      state.users[action.payload.index].password = action.payload.password;
+    },
+    setUserConfirmPasswordByIndex: (
+      state,
+      action: PayloadAction<UserConfirmPasswordPayload>
+    ) => {
+      state.users[action.payload.index].confirmPassword =
+        action.payload.confirmPassword;
+    },
+    setUserSshKedByIndex: (state, action: PayloadAction<UserSshKeyPayload>) => {
+      state.users[action.payload.index].ssh_key = action.payload.sshKey;
+    },
+
+    setUserAdministratorByIndex: (
+      state,
+      action: PayloadAction<UserAdministratorPayload>
+    ) => {
+      state.users[action.payload.index].isAdministrator =
+        action.payload.isAdministrator;
+      if (action.payload.isAdministrator) {
+        state.users?.[action.payload.index].groups?.push('wheel');
+      } else {
+        state.users[0].groups = state.users?.[0].groups?.filter(
```

Comment:
I don't think we need to hard code a 0 index here since the index comes in with the payload.